packages/core/lib/modules/accounts.js

```diff
@@ -2,63 +2,126 @@ import {WalletRpc} from '../client/wallet-rpc';
+  accountUpdated: 'account-updated',
+  accountRemoved: 'account-removed',
+};
 
 // TODO: Add events to accounts
 export class Accounts {
   accounts: Account[];
   wallet: Wallet;
+  eventManager: EventManager;
+  static DocumentFilters = {
+    mnemonicType: (item: WalletDocument) => item.type === 'Mnemonic',
+    currencyType: (item: WalletDocument) => item.type === 'Currency',
+  };
 
   constructor() {
     this.accounts = [];
     this.wallet = Wallet.getInstance();
+    this.eventManager = new EventManager();
+    this.eventManager.registerEvents(AccountsEvents);
   }
 
   async load() {
     this.accounts = await Wallet.getInstance().query({
-      type: 'account',
+      type: 'Address',
     });
-
+    this.eventManager.emit(AccountsEvents.loaded);
     return this.accounts;
   }
 
-  getBalance(address) {
-    return ApiRpc.getAccountBalance(accountId);
+  async exportAccount(accountId, password) {
+    return WalletRpc.exportAccount(accountId, password);
+  }
+  
+  async importAccount(json, password) {
+    return this.create({
+      json,
+      password,
+    })
+  }
+
+  async fetchBalance(accountId) {
+    const account = await this.getAccount(accountId);
+    const balance = await ApiRpc.getAccountBalance(account.address);
+    const currency = await this.findCorrelationByType(accountId, 'Currency');
+    
+    currency.value = balance;
+    
+    await this.wallet.update(currency);
+   
+    return balance;
+  }
+  
+  async getBalance(accountId) {
+    const currency = await this.findCorrelationByType(accountId, 'Currency');
+    return currency.value;
   }
 
   getAccounts() {
     return this.accounts;
   }
 
+  async getAccount(accountId): Promise<Account> {
+    const accountDocument = await Wallet.getInstance().getDocumentById(
+      accountId,
+    );
+
+    return {
+      ...accountDocument,
+      correlations: await WalletRpc.resolveCorrelations(accountId),
+    };
+  }
+
+  async findCorrelationByType(accountId: string, type: DocumentType) {
+    const account = await this.getAccount(accountId);
+    return account.correlations.find(c => c.type === type);
+  }
+
   generateMnemonic(): Promise<string> {
     return UtilCryptoRpc.mnemonicGenerate(12);
   }
 
-  async create(params: {
-    name: string,
-    type: KeypairType,
-    derivationPath: string,
-    mnemonic: string,
-  } = {}): Promise<Account> {
-   
-    const name = params.name;
-    const type = params.type || 'sr25519';
-    const phrase = params.mnemonic || await this.generateMnemonic();
+  async update(account: Account) {
+    await WalletRpc.update(account);
+    this.eventManager.emit(AccountsEvents.accountUpdated);
+    await this.load();
+  }
+
+  async create(
+    params: {
+      name: string,
+      keyPairType: KeypairType,
+      derivationPath: string,
+      mnemonic: string,
+      json: string,
+      password: string,
+    } = {},
+  ): Promise<Account> {
+    let {name, json, password, keyPairType = 'sr25519'} = params;
```

Comment:
do we need some validation on any of these inputs?

Reply:
yes, will be handled later https://www.notion.so/dockteam/Wallet-sdk-code-review-25d2f2d7644c470ea5a1012a8cd7c871